src/main/java/org/springframework/retry/backoff/ExponentialBackOffPolicy.java

```diff
@@ -289,15 +289,27 @@ public synchronized long getSleepAndIncrement() {
 		}
 
 		protected long getNextInterval() {
-			return (long) (this.interval * getMultiplier());
+			if (initialIntervalSupplier == null) {
+				return (long) (this.interval * getMultiplier());
+			}
+			else {
+				return this.interval == DEFAULT_INITIAL_INTERVAL
+						? (long) (this.initialIntervalSupplier.get() * getMultiplier())
+						: (long) (this.interval * getMultiplier());
+			}
 		}
 
 		public double getMultiplier() {
 			return this.multiplierSupplier != null ? this.multiplierSupplier.get() : this.multiplier;
 		}
 
 		public long getInterval() {
-			return this.intervalSupplier != null ? this.intervalSupplier.get() : this.interval;
+			if (initialIntervalSupplier == null) {
```

Comment:
Need `this.` prefix for this property use

Reply:
fixed